Libs/MRML/Logic/vtkMRMLApplicationLogic.cxx

```diff
@@ -823,3 +824,68 @@ void vtkMRMLApplicationLogic::EditNode(vtkMRMLNode* node)
   // Observers in qSlicerCoreApplication listen for this event
   this->InvokeEvent(EditNodeEvent, node);
 }
+
+//----------------------------------------------------------------------------
+void vtkMRMLApplicationLogic::RegisterModuleLogic(const char* moduleName,
+                                                  vtkMRMLAbstractLogic* moduleLogic)
+{
+  if (!moduleName)
+    {
+    vtkErrorMacro("RegisterModuleLogic: invaild module name.");
+    return;
+    }
+
+  // NOTE: Should we accept registation of modules that have no logic (e.g. Welcome)
+  //       I think we should, and return nullptr on GetModuleLogic. This can be a good
+  //       way to indicate there is that module that has no logic, as opposed to, the module
+  //       is not registered (could be that the module does not exists at all).
+  // if (!moduleLogic)
+  //   {
+  //   vtkErrorMacro("RegisterModuleLogic: module logic.");
+  //   return;
+  //   }
+
+  // Check that the module has not any registered logic.
+  if (this->Internal->ModuleLogicMap.count(moduleName))
```

Comment:
```suggestion
if (this->Internal->ModuleLogicMap.count(moduleName) > 0)
```

Reply:
We should use the `empty()` method. In this case performance does not matter, but it is good practice and even for readability it is nicer to check if a container is empty instead of counting number of elements.

Reply:
```suggestion
  if (!this->Internal->ModuleLogicMap.empty())
```

Reply:
Actually, this line is not to check whether the map is empty, but to check whether `moduleName` exists in the map. `std::map::empty()` does not apply here.

Reply:
In this case, `count()` would be fine (count is guaranteed to be 1, so it will not look for additional items). In general, it is more efficient to call `count()` only if you actually need count and otherwise getting the first element is faster (as it stops after it has found one element).